datafusion/optimizer/src/analyzer/subquery.rs

```diff
@@ -338,3 +339,57 @@ fn check_mixed_out_refer_in_window(window: &Window) -> Result<()> {
         Ok(())
     }
 }
+
+#[cfg(test)]
```

Comment:
I recommend putting this test somewhere in https://github.com/apache/arrow-datafusion/tree/main/datafusion/core/tests/user_defined (mostly so it is easier to find and to ensure everything is working as expected)

I think you may be able to reuse the dummy user defined code there

Reply:
Great suggestion. I plan to cover this case in `datafusion/core/tests/user_defined/user_defined_plan.rs`. But since the previous TopK is based on optimizer, I may need to add another extension plan and an analyzer rule.